### ISSUE TITLE
Typegoose breaks ts-jest unit tests

### ISSUE THREAD

**Issue by AramZS** [szokodiakos/typegoose#220]
Latest updates to ts-jest, Typescript and Mongoose are breaking Typegoose for unclear reasons. No error is thrown, it just hangs. I understand the library is set only for compatibility with a particular version of Mongoose, but that version has its own problems. It would be nice to have an up to date version that maintains compatibility or at least throw a meaningful error.

**Comment by ctrlplusb** [comment]
Ah dang, I thought I was going crazy. Yeah, I am experiencing the same issue. Timeouts, even with the simplest of cases. Did you by any chance gain more insight into this?

**Comment by ctrlplusb** [comment]
Hey @AramZS, I figured this out. It had nothing to do with `ts-jest` or `typegoose` in my case. I am using the `mongodb-memory-server` for my tests and noted that the database connections were left hanging, so it appeared that my tests would time out. I needed to ensure I disconnect from my db after my tests have run, killing all existing connections.

In a test file:

```typescript
import * as db from '../index';

beforeAll(async () => {
  await db.startDB();
});

afterAll(async () => {
  // 👇very important
  await db.disconnect();
});
```

In the db helper file:

```typescript
export const disconnect = async () => {
  logger.verbose('Disconnecting db');
  // Typescript types for mongoose claim there is no "connections"
  // property on "mongoose", but there is.
  // @ts-ignore
  const { connections } = mongoose;
  // @ts-ignore
  connections.forEach(con => {
    if (con.db) {
      con.db.unref();
    }
    con.close();
  });
  await mongoose.connection.close();
  await mongoose.disconnect();
  await new Promise(resolve => {
    mongoose.connection.once('close', () => {
      resolve();
    });
  });
};
```

**Comment by hasezoey** [comment]
> I understand the library is set only for compatibility with a particular version of Mongoose

This can be closed, because typegoose is not tied to a specific mongoose version anymore.

@Ben305

**Comment by hasezoey** [comment]
This might be the same issue as #380, this one can be closed in favor of #380 

@Ben305
